spec/models/case/sar/offender_spec.rb

```diff
@@ -72,6 +72,17 @@
     end
   end
 
+  describe "#set_number" do
+    let(:case_rejected)  { create(:offender_sar_case, :rejected, received_date: Date.parse("08/02/2024")) }
+
+    it "is composed of prefix R, received date and an incremented suffix" do
```

Comment:
You're only testing the prefix, so the test name needs changing and you also don't need to set the received date or use timecop.

Also, there should be a test for when the case is not rejected.